python/oak/mapping_ros.py

```diff
@@ -0,0 +1,220 @@
+        positions = keyframe.pointCloud.getPositionData()
+        pc = np.zeros((positions.shape[0], 6), dtype=np.float32)
+        p_C = np.vstack((positions.T, np.ones((1, positions.shape[0])))).T
+        pc[:, :3] = (camToWorld @ p_C[:, :, None])[:, :3, 0]
+
+        msg = PointCloud2()
+        msg.header.stamp = rospy.Time.now()
+        msg.header.frame_id = "world"
+        if keyframe.pointCloud.hasColors():
+            pc[:, 3:] = keyframe.pointCloud.getRGB24Data() * (1. / 255.)
+        msg.point_step = 4 * 6
+        msg.height = 1
+        msg.width = pc.shape[0]
+        msg.row_step = msg.point_step * pc.shape[0]
+        msg.data = pc.tobytes()
+        msg.is_bigendian = False
+        msg.is_dense = False
+        ros_dtype = PointField.FLOAT32
+        itemsize = np.dtype(np.float32).itemsize
+        msg.fields = [PointField(name=n, offset=i*itemsize, datatype=ros_dtype, count=1) for i, n in enumerate('xyzrgb')]
+        self.point_publisher.publish(msg)
+
+
+def parseArgs():
+    import argparse
+    p = argparse.ArgumentParser(__doc__)
+    p.add_argument("--smooth", help="Apply some smoothing to 3rd person camera movement", action="store_true")
+    p.add_argument('--ir_dot_brightness', help='OAK-D Pro (W) IR laser projector brightness (mA), 0 - 1200', type=float, default=0)
+    p.add_argument("--useRectification", help="--dataFolder option can also be used with some non-OAK-D recordings, but this parameter must be set if the videos inputs are not rectified.", action="store_true")
+    p.add_argument('--map', help='Map file to load', default=None)
+    p.add_argument('--save-map', help='Map file to save', default=None)
+    return p.parse_args()
+
+if __name__ == '__main__':
+    args = parseArgs()
+
+    configInternal = {
+        "computeStereoPointCloud": "true",
+        "pointCloudNormalsEnabled": "true",
+        "computeDenseStereoDepth": "true",
+    }
+    if args.useRectification:
+        configInternal["useRectification"] = "true"
+    else:
+        configInternal["alreadyRectified"] = "true"
+
+    slam_node = SLAMNode()
+
+    def onVioOutput(vioOutput):
+        cameraPose = vioOutput.getCameraPose(0)
+        camToWorld = cameraPose.getCameraToWorldMatrix()
+        slam_node.newOdometryFrame(camToWorld)
+
+    def onMappingOutput(output):
+        for frame_id in output.updatedKeyFrames:
+            keyFrame = output.map.keyFrames.get(frame_id)
+
+            # Remove deleted key frames from visualisation
+            if not keyFrame:
+                continue
+
+            # Check that point cloud exists
+            if not keyFrame.pointCloud: continue
+
+            if not slam_node.has_keyframe(frame_id):
+                slam_node.newKeyFrame(frame_id, keyFrame)
+
+        if output.finalMap:
+            print("Final map ready!")
+
+    print("Starting OAK-D device")
+    pipeline = depthai.Pipeline()
+    config = spectacularAI.depthai.Configuration()
+    config.useColor = True
```

Comment:
This `useColor` parameter only works on OAK-D S2+ devices, and it would be good to add a note about this, e.g., to the docstring. An alternative would be adding a new parameter (e.g, `args.color`). Turning that off would enable supporting older OAK-D devices as well. However, that would also require some changes in `newKeyFrame`.